python/python/lance/dataset.py

```diff
@@ -337,6 +338,11 @@ def scanner(
             .use_stats(use_stats)
             .fast_search(fast_search)
         )
+        if full_text_query is not None:
+            if isinstance(full_text_query, str):
+                builder = builder.full_text_search(full_text_query)
+            else:
+                builder = builder.full_text_search(**full_text_query)
```

Comment:
I think, from looking at the rust code, the `full_text_search` method only accepts a `dict`.  In that case I don't think you need to pass in `**full_text_query` and can just pass in `full_text_query`.

Also, if the user provides a `str` won't the call on line 343 fail?

Reply:
The builder's method is:
```python
    def full_text_search(
        self,
        query: str,
        columns: Optional[List[str]] = None,
    ) -> ScannerBuilder:
        """
        Filter rows by full text searching. *Experimental API*,
        may remove it after we support to do this within `filter` SQL-like expression

        Must create inverted index on the given column before searching,
        """
        self._full_text_query = {"query": query, "columns": columns}
        return self
```
so it won't fail.